Robot2018/src/tests/scriptupload/scriptupload.py

```diff
@@ -1,28 +1,29 @@
 #nt setup
```

Comment:
This file appears to be untested and buggy, and it lacks unit tests. It should not be merged without them.

Reply:
I don't know how to write a unit test for a command. Where can I learn this?

Reply:
Since "command" could refer to either the Command class above or the python script below, I want to clarify that you should have tests for the parsing you are doing in Java and tests for the python code. On the python side, I'm not particularly worried about tests for how you handle interaction with the user. Instead, use [PyUnit](https://docs.python.org/3/library/unittest.html) to write one or more tests for a class (or method) that is supposed to take a filename and store it's contents in the appropriate Preferences key. Then write the code to make those tests pass.